fuzz/ada_c.c

```diff
@@ -10,52 +10,62 @@ int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
    * ada_c
    */
   ada_url out = ada_parse((char*)data, size);
+  
+  if (!out) {
+    return 0;
+  }
+
   bool is_valid = ada_is_valid(out);
+  
+  if (!is_valid) {
+    ada_free(out);
+    return 0;
+  }
```

Comment:
This is not ideal because if the result has failed to parse, we still want to continue to the remaining of the fuzzer. Particularly, if ada_parse fails, we don't test ada_can_parse

Reply:
Thank you for the feedback! I have updated the code to ensure that the fuzzer continues even if ada_parse fails or the URL is invalid. Instead of returning early the code now proceeds to test ada_can_parse